File: server/src/main/java/com/example/schoolsite/workWithDatabase/repo/SheduleRepository.java
package com.example.schoolsite.workWithDatabase.repo;

import com.example.schoolsite.entity.Shedule;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Date;

public interface SheduleRepository extends JpaRepository<Shedule, Long> {
    Shedule findByCalendarIdAndClassroomIDAndDateAndSubjectIDAndTeacherIDAndWeekDay(Long calendarId, Long classroomID, Date date, Long subjectID, Long teacherID, int weekDay);
    Shedule findByTeacherIDAndCalendarIdAndDate(Long teacherId, Long calendarId, Date date);
    Shedule findByCalendarIdAndClassroomIDAndDate(Long calendarId, Long classroomId, Date date);
}

File: server/src/main/java/com/example/schoolsite/workWithDatabase/repo/SheduleRepository.kt
package com.example.schoolsite.workWithDatabase.repo

import com.example.schoolsite.entity.Shedule
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Date
import java.util.List
import org.springframework.stereotype.Repository

@Repository
interface SheduleRepository : JpaRepository<Shedule?, Long?> {
    fun findByCalendarIdAndClassroomIDAndDateAndSubjectIDAndTeacherIDAndWeekDay(calendarId: Long?, classroomID: Long?, date: Date?, subjectID: Long?, teacherID: Long?, weekDay: Int): Shedule?
    fun findByTeacherIDAndCalendarIdAndDate(teacherId: Long?, calendarId: Long?, date: Date?): Shedule?
    fun findByCalendarIdAndClassroomIDAndDate(calendarId: Long?, classroomId: Long?, date: Date?): Shedule?
    fun findByDateAndClassroomIDAndSubjectID(date: Date?, classroomId: Long?, subjectId: Long?): Shedule?
    fun findAllByClassroomID(classroomId: Long?): List<Shedule?>?
    fun findAllByClassroomIDAndDate(classroomId: Long?, date: Date?): List<Shedule?>?
}